elastic/fleet-server
# updating upgrade_status: completed

## What is the problem this PR solves?

https://github.com/elastic/kibana/issues/139704

Updating `upgrade_status: completed` agent field when an upgrade is acked, this is a fix instead of kibana relying on `upgraded_at: null` field, which caused a Lucene bug.

## How does this PR solve the problem?

The `upgrade_status` is set to `started` initially by kibana, when the action is started. Fleet Server then has to set it to `completed`, to signal to kibana that the action has completed.


## How to test this PR locally

1. Enroll fleet server, and many agents with horde
2. Bulk select agents and upgrade immediately
3. Wait for upgrade to complete, notice that agents are gradually showing up as `Healthy` and the Upgrade callout decreases in count.
4. Expect that the lucene error does not come anymore on the `agents` API call



## Checklist





## Related issues
